internal/exec/engine.go

```diff
@@ -154,13 +154,25 @@ func (e *Engine) acquireConfig() (cfg types.Config, f resource.Fetcher, err erro
 // fetchProviderConfig returns the externally-provided configuration. It first
 // checks to see if the command-line option is present. If so, it uses that
 // source for the configuration. If the command-line option is not present, it
-// check's the engine's provider. An error is returned if the provider is
-// unavailable. This will also render the config (see renderConfig) before
+// checks for a user config in the system config dir. If that is also missing,
+// it checks the config engine's provider. An error is returned if the provider
+// is unavailable. This will also render the config (see renderConfig) before
 // returning.
 func (e Engine) fetchProviderConfig(f resource.Fetcher) (types.Config, error) {
-	cfg, r, err := cmdline.FetchConfig(f)
-	if err == providers.ErrNoProvider {
-		cfg, r, err = e.OEMConfig.FetchFunc()(f)
+	fetchers := []providers.FuncFetchConfig{
+		cmdline.FetchConfig,
+		system.FetchConfig,
```

Comment:
Should this and `e.OEMConfig.FetchFunc()` be swapped? Wouldn't this always override use supplied configs when there is a default config in the system dir?

Reply:
Erm, I think I got confused there.

The old way Ignition worked was to take the base config, append the given provider's base config, then append the user config (one of cmdline, a thing fetched by the provider, or a default config, prioritized in that order).

This now takes the base config, appends the provider's base config, then appends the user config (one of cmdline, user.ign from the initramfs, a provider fetched config, or a default config, prioritized in that order).

Is that correct or am I reading it incorrectly? Is there a reason to prioritize a oem partition user.ign over a provider fetched one?

Reply:
Looking at this further, it might be possible to refactor the [no supplied config case](https://github.com/bgilbert/ignition/blob/3571eabaf411020a4481214aff06bfa3b2fe144d/internal/exec/engine.go#L75) into this.

Reply:
This logic is looking for the first "online" provider, meaning that it will happily accept an empty file, or a script, or a cloud-config.  The no-supplied-config case is the opposite: we successfully read zero or more bytes of data from _some_ provider, but they weren't an Ignition config.  It seems cleaner to keep the two code paths separate.

Reply:
And your latter reading is correct.  My thinking was that if a user writes a `user.ign`, they are specifically asking for it to be used, so it should be a higher priority.